src/pipert2/core/base/pipe.py

```diff
@@ -73,17 +78,21 @@ def link(self, *wires):
             wires (Wire): List of wires to connect their routines
 
         """
-
         for wire in wires:
-            data_transmitter = wire.data_transmitter if wire.data_transmitter else self.default_data_transmitter
-
-            self.network.link(source=wire.source, destinations=wire.destinations, data_transmitter=data_transmitter)
+            self.wires.append(wire)
 
     def build(self):
         """Build the pipe to be ready to start working.
 
         """
 
+        wires_validator.validate_wires(self.wires)
```

Comment:
I like the idea of validating wires, but I think that it doesn't cover all cases.

Think about routines that were never mentioned in wires but were added to flow.
It'll cause what I call "floating" routines, and bugs will be hard to debug. 

We'll be happy to have a further discussion about this.

Reply:
You right, but I think the module that handles the wire validation shouldn't know about flows at all. 
I think Pipe class should contain 'validate_pipe' function that checks -> not floating routines and valid wires.
What do you think?